e2e/cypress/integration/messaging/channel_read_after_permalink_spec.js

```diff
@@ -0,0 +1,146 @@
+                    cy.document().then((doc) => {
+                        cy.spy(doc, 'execCommand').as('execCommandSpy');
+                    });
+
+                    // # click on permalink option
+                    cy.get(`#permalink_${postId}`).should('be.visible').click();
+
+                    // # copy permalink from button
+                    cy.get('#linkModalCopyLink').click();
+
+                    // # Copy link url from text area
+                    cy.get('#linkModalTextArea').invoke('val').as('linkText');
+
+                    cy.get('@execCommandSpy').should('have.been.calledWith', 'copy');
+
+                    // # close modal dialog
+                    cy.get('#linkModalCloseButton').click();
+                });
+            });
+        });
+
+        // # get current team id
+        cy.getCurrentTeamId().then((teamId) => {
+            // # create public channel to post permalink
+            cy.apiCreateChannel(teamId, channelName, channelName, 'O', 'Test channel').then((response) => {
+                const testChannel = response.body;
+
+                // # post the message on the channel
+                postMessageOnChannel(testChannel);
+
+                // # change user
+                cy.apiLogout();
+                cy.apiLogin('sysadmin');
+                cy.visit('/');
+
+                // # make sure for this user, unread channels are grouped
+                checkUnreadsGroupSeparately();
+
+                // # read the message and click the permalink
+                clickLink(testChannel);
+
+                // * check the channel is not under the unread channel list
+                cy.get('#unreadsChannelList').find('#sidebarItem_' + testChannel.name).should('not.exist');
+
+                // * check the channel is not marked as unread
+                cy.get('#sidebarItem_' + testChannel.name).invoke('attr', 'aria-label').should('not.include', 'unread');
+            });
+        });
+    });
+});
+
+function checkUnreadsGroupSeparately() {
+    // # click header area to get the dropdown menu
+    cy.get('#sidebarHeaderDropdownButton').click({force: true});
+    cy.wait(TIMEOUTS.TINY);
+
+    // # click account settings menu
+    cy.get('#accountSettings').find('button').click({force: true});
+    cy.wait(TIMEOUTS.TINY);
+
+    // # click on sidebar submenu
+    cy.get('#sidebarButton').click({force: true});
+    cy.wait(TIMEOUTS.TINY);
+
+    // # click on edit to edit the channel grouping
+    cy.get('#groupChannelsEdit').click({force: true});
+    cy.wait(TIMEOUTS.TINY);
+
+    // # check the option so unread channels are grouped
+    // * also make sure it is checked
+    cy.get('#unreadAtTopOption').check().should('be.checked');
+    cy.wait(TIMEOUTS.TINY);
+
+    // # save the configuration
+    cy.get('#saveSetting').click({force: true});
+    cy.wait(TIMEOUTS.TINY);
+
+    // # close the modal
+    cy.get('#accountSettingsHeader').find('.close').click({force: true});
+}
```

Comment:
Since we're not testing UI flow for changing sidebar settings, let's make this as custom Cypress API command, something like below [at this line](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/support/api_commands.js#L496):
```javascript
const defaultSidebarSettingPreference = {
    grouping: 'by_type',
    unreads_at_top: 'true',
    favorite_at_top: true,
    sorting: 'alpha',
};

/**
 * Saves theme preference of a user directly via API
 * This API assume that the user is logged in and has cookie to access
 * @param {Object} value - sidebar settings object.  Will pass default value if none is provided.
 */
Cypress.Commands.add('apiSaveSidebarSettingPreference', (value = JSON.stringify(defaultSidebarSettingPreference)) => {
    return cy.getCookie('MMUSERID').then((cookie) => {
        const preference = {
            user_id: cookie.value,
            category: 'sidebar_settings',
            name: '',
            value,
        };

        return cy.apiSaveUserPreference([preference]);
    });
});
```

Reply:
Thank you for this. Is there any reference I can look at for future similar issues? I have done on other PR the setting configuration in a similar way as before (moving all around the options menus), since I am not sure what is the signature of these server calls. If you could lead me to the reference, I can make the future test directly with this approach.

Reply:
For the reference, I think it might have been touched by my comment here - https://github.com/mattermost/mattermost-webapp/pull/4156#discussion_r344713744

Reply:
Yes, I saw it. Thank you very much!